mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/DeletedMessageWithContent.java

```diff
@@ -0,0 +1,53 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.vault;
+
+import java.io.InputStream;
+
+/**
+ * This class carries a {@link org.apache.james.vault.DeletedMessage}
+ * and its data inside an InputStream.
+ *
+ * The InputStream is created and maintained by the callers.
+ */
+public class DeletedMessageWithContent {
```

Comment:
this object should be Closable (or AutoClosable) to make it clear that we expect the using code to cleanup the resources it contains.

Reply:
I can make it become Closeable/AutoCloseable. But I can't manage those resources in the DeletedMessageZipper. Because variable are created outside of the class and passed as parameters. Then I think closing them should be the responsibility of callers

Reply:
Not really.

The Zipper should be closing InputStreams after fully copying them.

Please remember InputStream is mutable (and muted) there anyway.

Does it make sense to have the Zipper fully consuming streams but relying on the caller for closing them?